src/charm.py

```diff
@@ -196,7 +198,7 @@ def _root_certificate_matches_config(self) -> bool:
         if not self._config_ca_common_name:
             raise ValueError("CA common name should not be empty")
         ca_certificate_secret = self.model.get_secret(label=CA_CERTIFICATES_SECRET_LABEL)
-        ca_certificate_secret_content = ca_certificate_secret.get_content()
+        ca_certificate_secret_content = ca_certificate_secret.get_content(refresh=True)
```

Comment:
If I remember correctly this is not needed for the owner of the secret, no?
And is it needed anyway? when the config_changed event is handled we generate a new certificate, store it in a secret and we return, then when the secret_changed runs we should be able to read the content of the new secret, shouldn't we?

Reply:
from set_content:
```
content: A key-value mapping containing the payload of the secret,

Update the content of this secret.

This will create a new secret revision, and notify all units tracking the secret (the "observers") that a new revision is available with a
ops.SecretChangedEvent <ops.charm.SecretChangedEvent>.
```

from get_content:
```
Get the secret's content.

Returns:
    A copy of the secret's content dictionary.

Args:
    refresh: If true, fetch the latest revision's content and tell
        Juju to update to tracking that revision. The default is to get the content of the currently-tracked revision.
```

From these it reads like it's required, but not really explicitly. Notifying units doesn't necessarily mean it updated the value, and "tell juju to update to tracking that revision" doesn't mean anything.

I'm fine with it either way as long as it works.

Reply:
Yes this is needed, without the refresh the ca certificate would remain the old one.

Reply:
I remember this causing integration test failures when we validated [TLSENG-80 ](https://warthogs.atlassian.net/browse/TLSENG-80). But if it works now then it's fine for me.